### PR TITLE
Win32: add test of native a11y tree nodes

### PR DESCRIPTION
**Pull request by cbracken**
Adds AddSemanticsNodeUpdateWithChildren, which tests that the native
tree of IAccessible COM objects is an accurate representation of the
platform-agnostic accessibility tree.

Issue: https://github.com/flutter/flutter/issues/77838

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/windows/flutter_windows_view_unittests.cc [hunk 1]
```diff
@@ -209,12 +209,216 @@ TEST(FlutterWindowsEngine, AddSemanticsNodeUpdate) {
...
+  varrole.vt = VT_I4;
+  ASSERT_EQ(node0_accessible->get_accRole(varchild, &varrole), S_OK);
+  EXPECT_EQ(varrole.lVal, ROLE_SYSTEM_GROUPING);
+
+  // Verify child count.
+  long node0_child_count = 0;
+  ASSERT_EQ(node0_accessible->get_accChildCount(&node0_child_count), S_OK);
+  EXPECT_EQ(node0_child_count, 2);
+
+  {
```
Comment (chunhtai) [inline shell/platform/windows/flutter_windows_view_unittests.cc:303]: why additional scope?

File: shell/platform/windows/flutter_windows_view_unittests.cc [hunk 2]
```diff
@@ -209,12 +209,216 @@ TEST(FlutterWindowsEngine, AddSemanticsNodeUpdate) {
...
+    std::string value(_com_util::ConvertBSTRToString(bvalue));
+    EXPECT_EQ(value, "Uji");
+
+    // Verify node type is static text.
+    VARIANT varrole{};
+    varrole.vt = VT_I4;
+    ASSERT_EQ(node3_accessible->get_accRole(varchild, &varrole), S_OK);
+    EXPECT_EQ(varrole.lVal, ROLE_SYSTEM_STATICTEXT);
+
+    // Verify the parent node is the root.
```
Comment (chunhtai) [inline shell/platform/windows/flutter_windows_view_unittests.cc:411]: node 2?

File: shell/platform/windows/flutter_windows_view_unittests.cc (hunk 1 above)
Comment (cbracken) [reply shell/platform/windows/flutter_windows_view_unittests.cc:303]: Just to keep from shadowing variables (the variables for reading name/value/role).

File: shell/platform/windows/flutter_windows_view_unittests.cc (hunk 2 above)
Comment (cbracken) [reply shell/platform/windows/flutter_windows_view_unittests.cc:411]: Ouch good catch!

File: shell/platform/windows/flutter_windows_view_unittests.cc (hunk 1 above)
Comment (chunhtai) [reply shell/platform/windows/flutter_windows_view_unittests.cc:303]: this looks good :), was just wondering whether this is part of the test.

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**